lib/internal/bootstrap/node.js

```diff
@@ -319,6 +319,27 @@ ObjectDefineProperty(process, 'features', {
 }
 
 const { emitWarning, emitWarningSync } = require('internal/process/warning');
+const { getOptionValue } = require('internal/options');
+
+let kTypeStrippingMode = null;
+ObjectDefineProperty(process.features, 'typescript', {
+  __proto__: null,
+  get() {
+    if (kTypeStrippingMode === null) {
+      if (getOptionValue('--experimental-transform-types')) {
+        kTypeStrippingMode = 'transform';
+      } else if (getOptionValue('--experimental-strip-types')) {
+        kTypeStrippingMode = 'strip';
+      } else {
+        kTypeStrippingMode = undefined;
+      }
+    }
+    return kTypeStrippingMode;
+  },
+  configurable: false,
```

Comment:
All other `process.features` properties are configurable. Any reasons to make this not configurable?

```suggestion
  configurable: true,
```